scylla/src/client/pager.rs

```diff
@@ -699,7 +699,10 @@ impl QueryPager {
             ..Default::default()
         };
 
-        let load_balancing_policy = Arc::clone(&execution_profile.load_balancing_policy);
+        let load_balancing_policy = statement
+            .get_load_balancing_policy()
+            .map(Arc::clone)
+            .unwrap_or_else(|| Arc::clone(&execution_profile.load_balancing_policy));
```

Comment:
:question: Does this work?

Reply:
```suggestion
            .unwrap_or_else(|| &execution_profile.load_balancing_policy)
            .map(Arc::clone);
```

Reply:
> This would work:
> 
> ```rust
>         let load_balancing_policy = Arc::clone(
>             statement
>                 .get_load_balancing_policy()
>                 .unwrap_or_else(|| &execution_profile.load_balancing_policy),
>         );
> ```
> 
> Which one do you prefer?

I like the version you proposed the most.